packages/components/radio/src/RadioGroup.stories.tsx

```diff
@@ -1,52 +1,141 @@
-import { Meta, StoryObj } from '@storybook/react';
+import { Meta, ReactRenderer, StoryObj } from '@storybook/react';
 import { action } from '@storybook/addon-actions';
 import RadioGroup from './RadioGroup';
 import { useState } from 'react';
 import { expect, userEvent, within } from '@storybook/test';
+import { PartialStoryFn } from 'storybook/internal/types';
+import { RadioGroupProps } from './radio.types';
+import { ThemeProvider } from '@douro-ui/react';
+import { EmotionJSX } from '@emotion/react/types/jsx-namespace';
 
-const meta: Meta<typeof RadioGroup> = {
+const meta: Meta<RadioGroupProps> = {
   title: 'Example/RadioGroup',
   component: RadioGroup,
+  decorators: [
+    (
+      Story: PartialStoryFn<ReactRenderer, RadioGroupProps>,
+    ): EmotionJSX.Element => (
+      <ThemeProvider>
+        <Story />
+      </ThemeProvider>
+    ),
+  ],
   parameters: {
     layout: 'centered',
   },
   tags: ['autodocs'],
-};
+  args: {
+    disabled: false,
+    options: [
+      { label: 'Label 1', value: 'label1' },
+      { label: 'Label 2', value: 'label2' },
+      { label: 'Label 3', value: 'label3', disabled: true },
+      {
+        label: 'Label 4',
+        value: 'label4',
+        disabled: true,
+        checked: true,
+      },
+    ],
+    selectedValue: '',
+    onChange: (value: string) => action('circle-group-changed')(value),
+  },
+  argTypes: {
+    disabled: {
+      control: 'boolean',
+    },
+  },
+} satisfies Meta<RadioGroupProps>;
 
 export default meta;
 
-type Story = StoryObj<typeof meta>;
-
-const RadioGroupWithHooks = () => {
-  const [selected, setSelected] = useState('');
-
-  const handleChange = (value: string) => {
-    action('circle-group-changed')(value);
-    setSelected(value);
-  };
-
-  return (
-    <RadioGroup
-      name="circleRadioGroup"
-      options={[
-        { label: 'Circle Radio 1', value: 'circle1' },
-        { label: 'Circle Radio 2', value: 'circle2' },
-        { label: 'Circle Radio 3', value: 'circle3', disabled: true },
-      ]}
-      selectedValue={selected}
-      onChange={handleChange}
-    />
-  );
+type Story = StoryObj<RadioGroupProps>;
+
+export const Default: Story = {
+  args: {
+    disabled: false,
+    options: [
+      { label: 'Circle Radio 1', value: 'label1' },
+      { label: 'Circle Radio 2', value: 'label2' },
+      { label: 'Circle Radio 3', value: 'label3' },
+      {
+        label: 'Circle Radio 4',
+        value: 'label4',
+      },
+    ],
+  },
+  render: (args: RadioGroupProps) => {
+    const [selected, setSelected] = useState('');
+
+    const handleChange = (value: string) => {
+      action('circle-group-changed')(value);
+      setSelected(value);
+    };
+
+    return (
+      <RadioGroup {...args} selectedValue={selected} onChange={handleChange} />
```

Comment:
Args to the end

Reply:
It's not possible because the Storybook needs the args before in order to be able to manipulate and control the behavior of the component effectively.